### PR TITLE
add validation to s9pks

### LINKED ISSUE #1149: [feat]: validate docker_image tags on s9pk load
we need to make sure packages cannot import images outside their namespace & version
we need to reverse engineer the image.tar format in order to do this

### PR DESCRIPTION
**Pull request by dr-bonez**
closes #1149 

also validates manifest for internal consistency

### CODE REVIEW

**Comment by dr-bonez** [comment]
@chrisguida @elvece for an improved developer experience, include `embassy-sdk verify <id>.s9pk` into your make process

**Comment by chrisguida** [comment]
Do you need us to test this before merge?

**Comment by dr-bonez** [comment]
closes #1163

File: backend/install-sdk.sh [hunk 1]
```diff
@@ -3,9 +3,9 @@
 set -e
 shopt -s expand_aliases
 
-if [ "$0" != "./build-sdk.sh" ]; then
+if [ "$0" != "./install-sdk.sh" ]; then
 	>&2 echo "Must be run from backend directory"
 	exit 1
 fi
 
-cargo install --bin=embassy-sdk --path=. --no-default-features --verbose
```
Comment (Blu-J) [inline backend/install-sdk.sh:11]: No more verbose?

File: backend/src/action/docker.rs [hunk 2]
```diff
@@ -18,11 +18,22 @@ use crate::id::{Id, ImageId};
...
+lazy_static::lazy_static! {
+    pub static ref SYSTEM_IMAGES: BTreeSet<ImageId> = {
+        let mut set = BTreeSet::new();
+
+        set.insert("compat".parse().unwrap());
+        set.insert("utils".parse().unwrap());
+
+        set
+    };
+}
```
Comment (Blu-J) [inline backend/src/action/docker.rs:35]: This would be okay to be a macro

File: backend/install-sdk.sh (hunk 1 above)
Comment (dr-bonez) [reply backend/install-sdk.sh:11]: i never saw the purpose of it. just a bunch of rustc output

File: backend/src/action/docker.rs (hunk 2 above)
Comment (dr-bonez) [reply backend/src/action/docker.rs:35]: ideally I'd like to generated this dynamically based on the contents of the `system-images` dir, but not worrying about that rn

File: backend/src/action/docker.rs [hunk 3]
```diff
@@ -44,6 +55,32 @@ pub struct DockerAction {
     pub sigterm_timeout: Option<SerdeDuration>,
 }
 impl DockerAction {
+    pub fn validate(
```
Comment (Blu-J) [inline backend/src/action/docker.rs:58]: Opinion: We could have had this return a Validated<DockerAction>. That would make things that need the validation to be run to know that this function was ran.

**Review by Blu-J — APPROVED** [review]

File: backend/src/developer/mod.rs [hunk 4]
```diff
@@ -28,3 +28,8 @@ pub fn init(#[context] ctx: SdkContext) -> Result<(), Error> {
     }
     Ok(())
 }
+
+#[command(subcommands(crate::s9pk::verify, crate::config::verify_spec))]
+pub fn verify() -> Result<(), Error> {
+    Ok(())
```
[the remark this answers is not in the document]
Comment (dr-bonez) [reply backend/src/developer/mod.rs:34]: this is a parent rpc command. it exists only to provide the subcommands `verify s9pk` and `verify config-spec`

**Pull request merged**